ledger/participant-integration-api/src/main/scala/platform/store/dao/events/SqlFunctions.scala

```diff
@@ -62,8 +62,31 @@ private[dao] object SqlFunctions {
   object OracleSqlFunctions extends SqlFunctions {
     // TODO https://github.com/digital-asset/daml/issues/9493
     // This is likely extremely inefficient due to the multiple full tablescans on unindexed varray column
-    override def arrayIntersectionWhereClause(arrayColumn: String, parties: Set[Party]): String =
-      s"""JSON_EXISTS($arrayColumn, '$$[*]?(@ in ("${parties.mkString("""", """")}"))')"""
+    override def arrayIntersectionWhereClause(arrayColumn: String, parties: Set[Party]): String = {
+      val NumCharsBetweenParties = 3
```

Comment:
Is it worth actually parsing to get character lengths from the live expression on line 86 so these checks don't rot if the expression is changed in future?